core/server/common/src/main/java/alluxio/master/journal/raft/BufferedJournalApplier.java

```diff
@@ -309,5 +310,20 @@ protected void runCatchup() {
         }
       }
     }
+
+    @Override
+    public void onError(Throwable t) {
+      if (ExceptionUtils.containsInterruptedException(t)) {
+        // Tolerate interruption when the master is failing over or closing
+        // so we don't extra-crash
+        LOG.warn("Thread {} interrupted, assume the master is failing over or shutting down",
+            Thread.currentThread().getId());
+        return;
+      }
+      LOG.error("Uncaught exception from thread {}", Thread.currentThread().getId(), t);
+      setError(t);
+      // An exception here means journal replay has failed and should crash the standby master
+      ProcessUtils.fatalError(LOG, t, "Failed to catch up journal");
```

Comment:
we are not 100% confident that current shutdown hooks shuts down the alluxio master correctly. Instead of throwing a fatal error, we can take an intermediate step to just log the error.

Reply:
Yea I agree logging is always a good 1st step if we are not sure about more steps.

@tcrain @jenoudet @yuzhu if any of you are also hesitant about just crashing, I will merge only the logging part in our 2.9 so at least we take a baby step forward.

Reply:
I think `System.exit()` calls should be avoided in favor of error propagation. If the error is severe enough to crash the master, then it should be an `AlluxioRuntimeException`.

Reply:
good call, i guess this part belongs to the scope of https://github.com/Alluxio/alluxio/issues/16381 and I will postpone resolving that until we've at least talked about it once and reached some level of agreement. I'll probably draft some code and bring it up by then.